app/controllers/tpi/publications_controller.rb

```diff
@@ -1,16 +1,61 @@
 module TPI
   class PublicationsController < TPIController
+    # rubocop:disable Metrics/AbcSize
     def index
-      publications = Publication.order(publication_date: :desc)
-      news = NewsArticle.order(publication_date: :desc)
+      @tags = (
```

Comment:
I would move that code into service object.

Also, without API we could do it differently:
- When request format is js. `format.js` just return html with filtered list and replace list html on the page. I guess that would work just fine.

Reply:
I can try to rewrite it a little bit